src/primitive_core.cc

```diff
@@ -68,17 +72,88 @@ namespace toit {
 
 MODULE_IMPLEMENTATION(core, MODULE_CORE)
 
-PRIMITIVE(write_string_on_stdout) {
-  ARGS(cstring, message, bool, add_newline);
-  fprintf(stdout, "%s%s", message, add_newline ? "\n" : "");
-  fflush(stdout);
+#if defined(TOIT_WINDOWS)
+#include <windows.h>
+#include <vector>
+
+static void replace_line_endings(const uint8_t* bytes, size_t length, std::vector<uint8_t>& buffer) {
```

Comment:
I'd probably use `uint8` instead of the _t variants in here.

```suggestion
static void replace_line_endings(const uint8* bytes, size_t length, std::vector<uint8>& buffer) {
```

Reply:
done.